actionpack/lib/action_controller/metal/head.rb

```diff
@@ -17,8 +17,13 @@ module Head
     #
     # See Rack::Utils::SYMBOL_TO_STATUS_CODE for a full list of valid +status+ symbols.
     def head(status, options = {})
-      options, status = status, nil if status.is_a?(Hash)
-      status ||= options.delete(:status) || :ok
+      if status.is_a?(Hash)
+        msg = status[:status] ? 'The :status option' : 'The implicit :ok status'
+        options, status = status, status.delete(:status) { :ok }
+
+        ActiveSupport::Deprecation.warn(msg + ' on `head` has been deprecated and will be removed in Rails 5.1. Please pass the status as a separate parameter before the options, instead.')
```

Comment:
We could use `squish` here to great effect: https://github.com/rails/rails/blob/3179b4a868c370bf879c15c53b78f25fadec9b41/activerecord/lib/active_record/relation/predicate_builder/class_handler.rb#L19-L23